[WIP] Add ddmrp from Eficent repository

guewen opened the pull request:
* Extract the addon from https://github.com/Eficent/ddmrp
* Migrate ddmrp to version 11.0
* Note: working on https://github.com/Eficent/ddmrp/pull/47 until ready, then I'll update this one for inclusion in OCA

The other addons of the repository will need to be extracted the same way.

Require #2 for the build

JordiBForgeFlow committed:
Initial commit

JordiBForgeFlow committed:
[IMP] Major improvements. This is rolling!!

JordiBForgeFlow committed:
[IMP] add security rules

JordiBForgeFlow committed:
[FIX]  wrong xml data

JordiBForgeFlow committed:
[FIX] change method to create procurements from buffers

JordiBForgeFlow committed:
[FIX] pass extra data to procurement

JordiBForgeFlow committed:
[FIX] various fixes and improvements

JordiBForgeFlow committed:
[FIX] various fixes

JordiBForgeFlow committed:
[FIX] various fixes

JordiBForgeFlow committed:
[FIX] various fixes and improvements

JordiBForgeFlow committed:
[FIX] various fixes and improvements

JordiBForgeFlow committed:
[IMP] split demand estimate to a separate module

JordiBForgeFlow committed:
[FIX] update readme

JordiBForgeFlow committed:
[FIX] filter execution priority level only on open manufacturing orders

JordiBForgeFlow committed:
[IMP] search optimization

JordiBForgeFlow committed:
[FIX] fix search

JordiBForgeFlow committed:
[FIX] search optimization

JordiBForgeFlow committed:
[FIX] fix search optimization

JordiBForgeFlow committed:
[FIX] search optimization

JordiBForgeFlow committed:
[FIX]  corrected issue with wizard

JordiBForgeFlow committed:
[FIX] update stock_demand_estimate

JordiBForgeFlow committed:
[FIX] alignment

JordiBForgeFlow committed:
[FIX] update stock_orderpoint_uom to align to PR to OCA

JordiBForgeFlow committed:
[IMP] add modules 'mrp_production_service' and 'procurement_service'

JordiBForgeFlow committed:
[IMP] add tests

JordiBForgeFlow committed:
[FIX] fix buffer view

JordiBForgeFlow committed:
[FIX] access to open uncompleted moves

JordiBForgeFlow committed:
[IMP] improve tests

JordiBForgeFlow committed:
[IMP] various improvements

JordiBForgeFlow committed:
[FIX]

JordiBForgeFlow committed:
Migration to v9

JordiBForgeFlow committed:
[ADD] stock_warehouse_orderpoint_stock_info_unreserved

JordiBForgeFlow committed:
[IMP] migrated to 9.0 "stock_warehouse_orderpoint_stock_info"

JordiBForgeFlow committed:
[FIX]

JordiBForgeFlow committed:
fix

JordiBForgeFlow committed:
fix

JordiBForgeFlow committed:
improve compute_orderpoint

JordiBForgeFlow committed:
[FIX] internal moves are not considered demand

JordiBForgeFlow committed:
add module ddmrp_exclude_moves_adu_calc

JordiBForgeFlow committed:
update

LoisRForgeFlow committed:
[9.0][ADD] ddmpr_mrp_production_request

JordiBForgeFlow committed:
fix incorrect procure_recommended_qty

LoisRForgeFlow committed:
[9.0][FIX] subtract qty only if positive.
Sometimes procurements are over-fulfilled so the subtract qty becomes negative, we don't want to apply it in that case, since we will be taking into account some stock moves twice.

LoisRForgeFlow committed:
[9.0][FIX] ddmrp_mrp_production_request: not passing buffer status from MR to MOs

LoisRForgeFlow committed:
[9.0] ddmrp: able to sort by priority in manufacturing orders.

LoisRForgeFlow committed:
[9.0][IMP] ddmrp: enhance extensibility of cron_ddmrp.

LoisRForgeFlow committed:
[9.0][IMP] ddmrp_mrp_production_request: able to sort by exectution priority.

LoisRForgeFlow committed:
[9.0][FIX] ddmrp: fix MO filters

JordiBForgeFlow committed:
Merge pull request #10 from Eficent/9.0-imp-ddmpr-mrp-sort-by-priority

[9.0] ddmrp: able to sort by priority in MOs and MRs.

LoisRForgeFlow committed:
[9.0][ADD] ddmrp_history

JordiBForgeFlow committed:
add ddmrp chart

LoisRForgeFlow committed:
cleanup

LoisRForgeFlow committed:
Merge pull request #12 from Eficent/9.0-web_kanban_ddmrp

[9.0] add ddmrp chart

LoisRForgeFlow committed:
[9.0][IMP] import exceptions

LoisRForgeFlow committed:
avoid uncomputable chart

JordiBForgeFlow committed:
fix orderpoint calc method in manufacturing requests and orders

LoisRForgeFlow committed:
[9.0][IMP] make DLT computed

LoisRForgeFlow committed:
[9.0][FIX] ddmrp: wrong on-hand status

LoisRForgeFlow committed:
[9.0][IMP] ddmrp: add refresh button to buffers

LoisRForgeFlow committed:
[9.0][IMP] ddmrp: split ADU computations into a new cron job.

LoisRForgeFlow committed:
[9.0][FIX] ddmrp_mrp_production_request: orderpoint computation.

LoisRForgeFlow committed:
[9.0][ADD] ddmrp_adjustment

LoisRForgeFlow committed:
[9.0][ADD] ddmrp_adjustment

LoisRForgeFlow committed:
[9.0][FIX] ddmrp_history: permission issue

JordiBForgeFlow committed:
add demo data

JordiBForgeFlow committed:
add location to demo data

JordiBForgeFlow committed:
improve bom structure report

JordiBForgeFlow committed:
move dlt logic to bom

JordiBForgeFlow committed:
various improvements

JordiBForgeFlow committed:
dlt: several improvements

JordiBForgeFlow committed:
fix

JordiBForgeFlow committed:
fix

JordiBForgeFlow committed:
add mrp_mts_mto demo data

LoisRForgeFlow committed:
[9.0][FIX] ddmrp_history: ondelete cascade

LoisRForgeFlow committed:
[FIX] ddmrp_history: wrong depiction

LoisRForgeFlow committed:
[FIX] ddmrp: wrong string

LoisRForgeFlow committed:
[9.0][FIX] ddmrp: ignore reservations

LoisRForgeFlow committed:
[9.0][IMP] ddmrp: enhance performace

LoisRForgeFlow committed:
[9.0][IMP] ddmrp_mrp_production_request: enhance performace

LoisRForgeFlow committed:
[9.0][FIX] ddmrp: fix infinite recursion

LoisRForgeFlow committed:
[9.0] ddmrp: addd ddmrp specific config menu

LoisRForgeFlow committed:
[9.0][ADD] ddmrp_product_replace

LoisRForgeFlow committed:
[9.0][IMP] ddmrp: add execution status to PO lines.

LoisRForgeFlow committed:
[9.0][FIX] ddmrp_mrp_production_request: fix infinite recursion

LoisRForgeFlow committed:
PO reference clickable

LoisRForgeFlow committed:
[9.0][IMP] ddmrp: enhance PO lines view

LoisRForgeFlow committed:
[9.0][REM] stock_orderpoint_uom: moved to OCA/stock-logistics-warehouse

LoisRForgeFlow committed:
[9.0][REM] stock_demand_estimate: moved to OCA/stock-logistics-warehouse

LoisRForgeFlow committed:
[9.0][REM] stock_warehouse_orderpoint_stock_info and stock_warehouse_orderpoint_stock_info_unreserved

LoisRForgeFlow committed:
[ADD] travis CI

LoisRForgeFlow committed:
[10.0][MIG] ddmrp

LoisRForgeFlow committed:
[10.0][MIG] ddmrp

LoisRForgeFlow committed:
[9.0][FIX] ddmrp: consider past due moves.

LoisRForgeFlow committed:
[9.0][REM] mrp_production_service

LoisRForgeFlow committed:
[REM] .idea files

LoisRForgeFlow committed:
[UPD] .gitignore

LoisRForgeFlow committed:
[UPD] .gitignore

LoisRForgeFlow committed:
[10.0][MIG] ddmrp_exclude_moves_adu_cal

LoisRForgeFlow committed:
[10.0][MIG] ddmrp_exclude_moves_adu_cal

LoisRForgeFlow committed:
[10.0][MIG] ddmrp_history

LoisRForgeFlow committed:
[10.0][MIG] ddmrp_mrp_production_request

LoisRForgeFlow committed:
[10.0][MIG] ddmrp_product_replace

LoisRForgeFlow committed:
[10.0][FIX] ddmrp fixes

hparfr committed:
[9.0][FIX] Use date_range intsead of period.id (#30)

florian-dacosta committed:
[9.0][FIX] ddmrp: Add wizard in init file (#31)

LoisRForgeFlow committed:
[9.0][FIX] readme and bom syntax regarding dlt computation

LoisRForgeFlow committed:
Merge pull request #13 from Eficent/9.0-imp-ddmrp-dlt

[9.0][IMP] make DLT computed

LoisRForgeFlow committed:
[9.0][IMP] ddmrp_adjustment: consider lead time on demand alocation for DAF

LoisRForgeFlow committed:
[9.0][IMP] ddmrp_adjustment: consider lead time on demand alocation for DAF

LoisRForgeFlow committed:
[9.0][IMP] ddmrp_adjustment: implement Lead Time Adj Factor

LoisRForgeFlow committed:
[9.0][IMP] ddmrp_adjustment: implement Lead Time Adj Factor

LoisRForgeFlow committed:
add setup file

LoisRForgeFlow committed:
Merge pull request #15 from Eficent/9.0-add-ddmrp_adjustment

[9.0][ADD] ddmrp_adjustment

LoisRForgeFlow committed:
[9.0] ddmrp: add setup files

LoisRForgeFlow committed:
[REM] .idea

LoisRForgeFlow committed:
[UPD] .gitignore

HviorForgeFlow committed:
[MIG] ddmrp_adjustment: Migration to 10.0

HviorForgeFlow committed:
[MIG] ddmrp_adjustment: Migration to 10.0

florian-dacosta committed:
[9.0][FIX] demand estimate domains (#34)

HviorForgeFlow committed:
setup.py

HviorForgeFlow committed:
setup.py

rgarnau committed:
fixes

rgarnau committed:
fixes

Commit:
Merge pull request #37 from Eficent/10.0-fix-setup

fixes

rgarnau committed:
[fix] setup

rgarnau committed:
[fix] setup

LoisRForgeFlow committed:
[9.0][ADD] setup files

LoisRForgeFlow committed:
Merge branch '10.0' into 10.0-mig-ddmrp

JordiBForgeFlow committed:
change supply qualification for net flow position to consider only supply
the moves that are within the DLT.

LoisRForgeFlow committed:
[10.0][MIG] ddmrp: bom structure report

LoisRForgeFlow committed:
fix travis

LoisRForgeFlow committed:
Merge pull request #25 from Eficent/10.0-mig-ddmrp

[10.0][FIX] ddmrp migration: add latest changes

Commit:
Simplify orderpoint_id fields on purchase and mrp.production

This is now a standard Many2one field in the purchase addon.
Keep the index though.

For mrp.production, we add the field which is filled on creation of the
manufacturing order.

HviorForgeFlow committed:
adding ddmrp buffer profiles data

Commit:
Migrate to model / api changes done in Odoo 11.0

* Move method override to the new wizard
  procure_calculation has been moved to a wizard working on procurement.group
  instead of procurement.order
* Adapt name of report model renamed in core mrp
* Use correct format in views for colors
* Update ir_cron xml to new structure
* Rename field product_uom_id on mrp.bom

LoisRForgeFlow committed:
[FIX] setup and travis

LoisRForgeFlow committed:
[FIX] setup and travis

LoisRForgeFlow committed:
[10.0][MIG] ddmrp_adjustment: migrate latest changes

LoisRForgeFlow committed:
[10.0][MIG] ddmrp_adjustment: migrate latest changes

Commit:
Remove procurement.order

Commit:
Fix typos in readme

Commit:
Fix tests

And use a SavepointCase which reduces the tests from ~10 seconds as the
setup is done only once

LoisRForgeFlow committed:
[10.0][FIX] ddmrp: enable the use of override buffers

grindtildeath committed:
[MIG] ddmrp_adjustment: Migration to 11.0

LoisRForgeFlow committed:
[9.0][FIX] ddmrp: issue with date on manually created pickings.

Commit:
[10.0][ADD] Execution history buffer (#49)

JordiBForgeFlow commented:
Thanks @guewen We're now busy preparing the training for next week, and we'll support you next week.
Please consider migrating this as well: https://github.com/Eficent/ddmrp/commit/a88a005d2e3aae1a4fd664243e58ce691dc94b21

LoisRForgeFlow committed:
[10.0][FIX] ddmrp: dlt computeation considers distributed parts.

guewen commented:
I set this PR as `Work in progress`, I'll continue the work on https://github.com/Eficent/ddmrp/pull/47 and update this one when ready. It'll ease the port of changes from other branches if it happens in the same repo.

Commit:
Compute adu value on create and write

The adu is computed by an onchange, but as the field is readonly, the
value is never passed to the server so doesn't change. Force the
computation on write and create.

Commit:
Use the buffer profile which match with the test data

hparfr committed:
Use expected_date instead of date in all cacl_qualified_deman

grindtildeath committed:
Ensure _calc_adu is called to play onchange on product_min/max_qty

grindtildeath committed:
Fix api decorators and add ensure_one()

HviorForgeFlow committed:
Merge pull request #55 from akretion/fix/date_expcted_in_calc_qualified_demand

[10][FIX] Use expected_date instead of date in all cacl_qualified_deman

rousseldenis committed:
[10.0][FIX] ddmrp: Change self to rec to avoid Singleton errors

pedrobaeza commented:
Please rebase to have CI and you can move WIP branch here for better tracking to all contributors.

LoisRForgeFlow commented:
@guewen there are some dependencies that aren't even proposed on a PR to v11 (as stock_demand_estimate), how are you working around that?

Commit:
Merge remote-tracking branch 'eficent/10.0' into mig-ddmrp

guewen commented:
@lreficent It is: https://github.com/OCA/stock-logistics-warehouse/pull/381 :)

LoisRForgeFlow commented:
@guewen yes, finally I found it (it has a name without underscores). I'm reviewing all the dependencies on OCA for ddmrp, I hope we could get everything merged ASAP and then move to review ddmrp itself! (the general overview is good :wink:)

grindtildeath committed:
Fix api decorator on create of mrp.production

grindtildeath committed:
Remove coding: utf-8 declaration from *.py files

grindtildeath committed:
Rename context key __calc_adu to __no_adu_calc

grindtildeath committed:
Remove dlt from buffer creation dicts in tests

LoisRForgeFlow committed:
[11.0][MIG] ddmrp:
* bump version
* remove unneded dependency
* fix small issues on views
* fix procure_recommended_qty computation

Commit:
Merge pull request #1 from Eficent/11.0-guewen

[11.0] ddmrp migration final fixes

Commit:
Fix lint

LoisRForgeFlow committed:
Merge pull request #47 from guewen/mig-ddmrp

Migration 11.0

grindtildeath committed:
[11.0][MIG] ddmrp_history (#52):

* Migration to 11.0
* Add python unittest
* Improve code according to reviews

grindtildeath committed:
[11.0][MIG] ddmrp_product_replace (#51)

* Migration to 11.0
* Add unittest
* Ensure putaway stategies are correctly copied

LoisRForgeFlow commented:
@guewen Everything is merged and ready at Eficent repo, would you update this? :wink:

LoisRForgeFlow committed:
[11.0][FIX] ADU *must* only be computed by the cron job (we do *not* want real-time ADU).

max3903 commented:
@guewen @lreficent Can we wait for https://github.com/Eficent/ddmrp/pull/60 to be merged?

LoisRForgeFlow commented:
@max3903 Yes, as you prefer :+1:

LoisRForgeFlow committed:
[11.0][IMP] ddmrp: more robust buffer status computation on PO lines.

StephanRozendaal committed:
[11.0]Fix travis config for 11.0

Fix the travis file for the tests, changes include:
 - update Python version to 3.5
 - Turn on tests
 - Update Odoo version to 11.0
 - Add temp web branch

StephanRozendaal committed:
Add requirements.txt

StephanRozendaal committed:
Set modules to installable = False

StephanRozendaal committed:
[REF] Separate compute function on mrp.bom

Separate the compute function _compute_is_buffered in two
functions. One function computes the orderpoint, and the other
one the the 'is_buffered' boolean.

StephanRozendaal committed:
[REF]Check product variant recordset

Check product variant recordset, before referencing them.

StephanRozendaal committed:
[REF] Add @api.depends on compute functions

Add @api.depends on compute functions so they are triggered by
field changes.

StephanRozendaal committed:
[IMP] Add tests for mrp.bom

Add tests to check if the BoM orderpoint and is_buffered are
properly set.

StephanRozendaal committed:
[IMP] Add tests for mrp.bom mto_rule

Add tests for the BoM to test if the mto_rule is set.

StephanRozendaal committed:
[IMP] Add tests for mrp.bom dlt

Add tests that check check various values of the dlt for a
BoM.

max3903 committed:
[FIX] KeyError: 'purchase_line_ids'

grindtildeath committed:
Use savepoints instead of manual commit and rollback

grindtildeath committed:
Fix W7908(missing-newline-extrafiles)

grindtildeath committed:
Fix [E7901(rst-syntax-error), ]  Enumerated list start value not ordinal-1: "B" (ordinal 2)

StephanRozendaal committed:
[REF] Remove explicit repo in oca_dependencies.txt

Remove the explicit repo URL for the manufacture repo since this is no longer
needed.

LoisRForgeFlow committed:
fix flake8 issues

grindtildeath committed:
Merge branch '11.0' into 11.0-mig-ddmrp-oca

grindtildeath committed:
Fix E7901(rst-syntax-error)

simahawk commented on ddmrp/README.rst:
```diff
@@ -0,0 +1,242 @@
+.. image:: https://img.shields.io/badge/license-AGPLv3-blue.svg
```
maybe switch to new readme structure?

LoisRForgeFlow replied:
We are still not ready to move to OCA unfortunately. We are still working on some important improvements you can follow here: https://github.com/Eficent/ddmrp/pull/69. But for sure, when the code is ready I will update the readme!

max3903 replied:
@lreficent Can we merge this one and when you are ready, you make your pull request with the corresponding version number?

simahawk reviewed (approved):
not reviewing everything since it has been already validated on Eficent repo, so I trust it's working.
Only remark: readme update if possible.

LoisRForgeFlow commented:
Closed in favour of https://github.com/OCA/ddmrp/pull/16. Look at the description for more details.
